src/auth/verification-code/verification-code.entity.ts

```diff
@@ -0,0 +1,20 @@
+import { User } from 'src/user/user.entity';
+import { Column, Entity, ManyToOne, PrimaryGeneratedColumn } from 'typeorm';
+
+@Entity({})
+export class VerificationCode {
+  @PrimaryGeneratedColumn()
+  id: number;
+
+  @Column()
+  code: string;
+
+  @Column()
+  validUntil: Date;
+
+  @Column()
+  used: boolean;
+
+  @ManyToOne(() => User, (user) => user.verificationCodes, { eager: true })
```

Comment:
instead of having a complete entity for verification codes, and keeping multiple docs per user (in case he clicks resned), why don't we just have a single field in the user entity representing the verification code and overwrite it when necessary ?

Reply:
Cuz it has some logic like when the code becomes invalid so I preferred to separate this in its own domain

Reply:
if it is overwritten it can't be found and hence invalid

Reply:
It might not have been overwritten but becomes invalid after 5 minutes. I mean the user can get one code and does not request a new one. After 5 minutes this code is invalid and is still there not overwritten